services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -622,6 +634,95 @@ public void cleanExcludes() {
+    assertContains(unrecognisedProperties, "testString");
+    assertEquals("test", routeOptions.getUnrecognizedProperty("testString").getAsString());
+    assertContains(unrecognisedProperties, "testNumber");
+    assertEquals(4.9, routeOptions.getUnrecognizedProperty("testNumber").getAsDouble(), 0);
+    assertContains(unrecognisedProperties, "testBoolean");
+    assertTrue(routeOptions.getUnrecognizedProperty("testBoolean").getAsBoolean());
+  }
+
+  @Test
+  public void unrecognizedOptionsFromJsonToUrl() throws IOException {
+    Gson gson = new GsonBuilder().create();
+    JsonObject routeOptionsJson = gson.fromJson(loadJsonFixture(ROUTE_OPTIONS_JSON), JsonObject.class);
+    routeOptionsJson.add("testString", new JsonPrimitive("string"));
+    routeOptionsJson.add("testBoolean", new JsonPrimitive(false));
+    routeOptionsJson.add("testInteger", new JsonPrimitive(88));
+    routeOptionsJson.add("testDouble", new JsonPrimitive(36.6));
+    JsonArray testArray = new JsonArray();
+    testArray.add(1);
+    testArray.add(2.5);
+    testArray.add(false);
+    routeOptionsJson.add("testArray", testArray);
+    JsonObject testObject = new JsonObject();
+    testObject.add("testField", new JsonPrimitive("a"));
+    routeOptionsJson.add("testObject", testObject);
+    RouteOptions routeOptions = RouteOptions.fromJson(routeOptionsJson.toString());
+
+    String query = routeOptions.toUrl("testToken").getQuery();
+    assertContains(query, "testString=string");
+    assertContains(query, "testBoolean=false");
+    assertContains(query, "testInteger=88");
+    assertContains(query, "testDouble=36.6");
+    assertDoesNotContain(query, "testArray");
+    assertDoesNotContain(query, "testObject");
+  }
+
+  @Test
+  public void allUnrecognisedPropertiesSurviveToFromJson() throws IOException {
+    Gson gson = new GsonBuilder().create();
+    JsonObject mutatedRouteOptionsJson = gson.fromJson(loadJsonFixture(ROUTE_OPTIONS_JSON), JsonObject.class);
+    MutateJsonUtil.mutateJson(mutatedRouteOptionsJson);
+
+    RouteOptions routeOptions = RouteOptions.fromJson(mutatedRouteOptionsJson.toString());
+    JsonObject deserializedRouteOptions = gson.fromJson(routeOptions.toJson(), JsonObject.class);
+
+    assertEquals(mutatedRouteOptionsJson, deserializedRouteOptions);
+  }
+
+  @Test
+  public void someUnrecognisedPropertiesDoNotSurviveToFromUrl() throws IOException {
+    Gson gson = new GsonBuilder().create();
+    JsonObject mutatedRouteOptionsJson = gson.fromJson(loadJsonFixture(ROUTE_OPTIONS_JSON), JsonObject.class);
+    MutateJsonUtil.mutateJson(mutatedRouteOptionsJson);
+
+    URL routeOptionsUrl = RouteOptions.fromJson(mutatedRouteOptionsJson.toString()).toUrl("testToken");
+    RouteOptions routeOptionsFromUrl = RouteOptions.fromUrl(routeOptionsUrl);
+    JsonObject deserializedRouteOptionsUrl = gson.fromJson(routeOptionsFromUrl.toJson(), JsonObject.class);
+
+    assertNotEquals(mutatedRouteOptionsJson, deserializedRouteOptionsUrl);
```

Comment:
Which properties didn't survive? The non-primitives? Again, I'd be in favor of just throwing right away instead of maintaining this disparity.